stat.c

```diff
@@ -3087,17 +3091,25 @@ int calc_log_samples(void)
 		if (!td->bw_log ||
 			(td->bw_log && !per_unit_log(td->bw_log))) {
 			tmp = add_bw_samples(td, &now);
-			if (tmp < next)
-				next = tmp;
+
+			if (td->bw_log)
+				log_avg_msec = td->bw_log->avg_msec;
 		}
 		if (!td->iops_log ||
 			(td->iops_log && !per_unit_log(td->iops_log))) {
 			tmp = add_iops_samples(td, &now);
-			if (tmp < next)
-				next = tmp;
+
+			if (td->iops_log)
+				log_avg_msec = td->iops_log->avg_msec;
 		}
+
+		if (tmp < next)
+			next = tmp;
 	}
 
+	next_mod = log_avg_msec==0 ? elapsed_time: elapsed_time%log_avg_msec;
```

Comment:
Please rewrite this using ```if``` instead, that's much easier to read. Also remember to have spaces between operators and variables. Goes for the next line, too.